modules/eks/cluster/CHANGELOG.md

```diff
@@ -1,4 +1,166 @@
-## Components PR [#852](https://github.com/cloudposse/terraform-aws-components/pull/852)
+## Components PR [#910](https://github.com/cloudposse/terraform-aws-components/pull/910)
+
+Bug fix and updates to Changelog, no action required.
+
+Fixed: Error about managed node group ARNs list being null, which could happen
+when adding a managed node group to an existing cluster that never had one.
+
+## Upgrading to `v1.303.0`
+
+Components PR [#852](https://github.com/cloudposse/terraform-aws-components/pull/852)
+
+This is a bug fix and feature enhancement update. No action is necessary to upgrade.
+However, with the new features and new recommendations, you may want to change
+your configuration.
+
+## Recommended (optional) changes
+
+Previously, we recommended deploying Karpenter to Fargate and not provisioning
+any nodes. However, this causes issues with add-ons that require compute power
+to fully initialize, such as `coredns`, and it can reduce the cluster to a
+single node, removing the high availability that comes from having a node
+per Availability Zone and replicas of pods spread across those nodes.
+
+As a result, we now recommend deploying a minimal node group with a single
+instance (currently recommended to be a `c6a.large`) in each of 3 Availability
+Zones. This will provide the compute power needed to initialize add-ons, and
+will provide high availability for the cluster. As a bonus, it will also
+remove the need to deploy Karpenter to Fargate.
```

Comment:
I don't know how current any recommendation to deploy Karpenter to Fargate is. I believe the point of the recommendation was to avoid the need for a managed node group, but we have found that a managed node group is worthwhile anyway. Karpenter has affinity/anti-affinity to prevent it from being deployed to nodes it manages, so that is not an issue. We recommend deploying a managed node group of 3 instances, so that should also not be a problem, as EKS should cycle the nodes one at a time and move workloads around to avoid outages. Plus Karpenter can be deployed redundantly (I recommend 2 replicas) to further reduce any chance of outages.